src/AppInstallerCLICore/PortableInstaller.h

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+#include "winget/PortableARPEntry.h"
+#include <filesystem>
+
+using namespace AppInstaller::Registry::Portable;
+
+namespace AppInstaller::CLI::Portable
+{
+
+    std::filesystem::path GetPortableLinksLocation(Manifest::ScopeEnum scope);
+
+    std::filesystem::path GetPortableInstallRoot(Manifest::ScopeEnum scope, Utility::Architecture arch);
+
+    // Object representation of the metadata and functionality required for installing a Portable package. 
+    struct PortableInstaller
+    {
+        std::string DisplayName;
+        std::string DisplayVersion;
+        std::string HelpLink;
+        std::string InstallDate;
+        std::filesystem::path InstallLocation;
+        std::filesystem::path PortableSymlinkFullPath;
+        std::filesystem::path PortableTargetFullPath;
+        std::string Publisher;
+        std::string SHA256;
+        std::string URLInfoAbout;
+        std::string UninstallString;
+        std::string WinGetInstallerType;
+        std::string WinGetPackageIdentifier;
+        std::string WinGetSourceIdentifier;
+        bool IsUpdate = false;
+        // If we fail to create a symlink, add install directory to PATH variable
+        bool InstallDirectoryAddedToPath = false;
+
+        PortableInstaller(Manifest::ScopeEnum scope, Utility::Architecture arch, const std::string& productCode);
+
+        HRESULT SingleInstall(const std::filesystem::path& installerPath);
+
+        HRESULT MultipleInstall(
+            const std::vector<Manifest::NestedInstallerFile>& nestedInstallerFiles,
+            const std::vector<std::filesystem::path>& extractedItems);
+
+        HRESULT Uninstall(bool purge = false)
+        {
+            if (std::filesystem::exists(GetPortableIndexPath()))
+            {
+                return UninstallFromIndex(purge);
+            }
+            else
+            {
+                return UninstallSingle(purge);
+            }
+        }
+
+        template<typename T>

```

Comment:
This name isn’t very descriptive to me. What is being committed and where. Perhaps adding something tying it to ARP Values would help